### PR TITLE
Add unit tests

### PR DESCRIPTION
**Pull request by ziqichen55555**
This pull request introduces unit tests for the PaperMate API, enhancing the overall test coverage of the application. The tests focus on the following aspects:
Mutation Tests: Validates the functionality of data mutations within the API, ensuring that the expected changes occur in the database.
Paper List Tests: Confirms the correct retrieval and processing of paper data, validating that the API endpoints return accurate information.
Search Functionality Tests: Tests the search features to ensure that queries yield expected results and handle edge cases properly.
These tests will help maintain the integrity of the application and catch potential issues early in the development process.
This PR is linked to issue [Test Suites #116](https://github.com/nitishragu12/Capstone-Project/issues/116).

### CODE REVIEW

File: docker-compose.yml
Comment (somni1oquist) [inline docker-compose.yml:1]: What's the necessity for this secret?

File: api/tests/test_mutation.py [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+    def tearDown(self):
+        db.session.remove()
+        db.drop_all()
+        self.app_context.pop()
+
+    def test_sample_data_inserted(self):
+        # Verify that data has been inserted into the database
+        paper = Paper.query.first()
+        self.assertIsNotNone(paper)
+        self.assertEqual(paper.doi, "10.37934/araset.48.1.137151")  # Make sure the inserted data is as expected
```
[the remark this answers is not in the document]
Comment (ziqichen55555) [reply api/tests/test_mutation.py:55]: I got difficulty here😓

[the remark this answers is not in the document]
Comment (somni1oquist) [reply api/tests/test_mutation.py]: Then add dummy code and label `# @TODO: assert mutation after request` for this PR's purpose.

File: api/tests/__init__.py
[the remark this answers is not in the document]
Comment (maxtse25) [reply api/tests/__init__.py:1]: Instead of overwriting, mock out specific components when necessary

File: api/tests/test_paper_list.py
[the remark this answers is not in the document]
Comment (maxtse25) [reply api/tests/test_paper_list.py:1]: I asked ChatGPT and it says `In your case, if performance isn’t a concern and you prefer test isolation (which is generally a good practice), switching to setUp and tearDown instance methods could be a better alternative.` lol

**Review by somni1oquist — CHANGES_REQUESTED** [review]
Please add coverage to .gitignore and recover docker-compose file

**Review by somni1oquist — APPROVED** [review]
Thanks for all the works!
